code/game/objects/items/crab17.dm

```diff
@@ -0,0 +1,208 @@
+/obj/item/suspiciousphone
+	name = "suspicious phone"
+	desc = "This device raises pink levels to unknown highs."
+	icon = 'icons/obj/items_and_weapons.dmi'
+	icon_state = "suspiciousphone"
+	w_class = WEIGHT_CLASS_SMALL
+	attack_verb = list("dumped")
+	var/dumped = FALSE
+
+/obj/item/suspiciousphone/attack_self(mob/user)
+	if(!ishuman(user))
+		to_chat(user, "<span class='warning'>This device is too advanced for you!</span>")
+		return
+	if(dumped)
+		to_chat(user, "<span class='warning'>You already activated Protocol CRAB-17.</span>")
+		return FALSE
+	if(alert(user, "Are you sure you want to crash this market with no survivors?", "Protocol CRAB-17", "Yes", "No") == "Yes")

```

Comment:
this is a sleep

Reply:
It is? What would a good alternative be then? I used alert code that was used in other places too and I'm not sure what a better solution would be.

Reply:
it's not that you ahve a better solution, it's that you HAVE TO CHECK IF THE ITEM IS QDELETED AFTER A SLEEP, EVERY TIME, ANY TIME, ALWAYS, 100% OF THE TIME

Reply:
right, i'll add a sanity check :)